### PR TITLE
Fix/logger flush

### PR DESCRIPTION
**Pull request by pgaultier**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes/no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

Linked to ticket created in https://github.com/yiisoft/log/issues/1

### CODE REVIEW

File: src/base/Application.php [hunk 1]
```diff
@@ -522,6 +522,8 @@ public function run()
 
             $this->state = self::STATE_END;
 
+            $this->flushLogger();
```
Comment (samdark) [inline src/base/Application.php]: Why flushing two times?

File: src/base/Application.php [hunk 2]
```diff
@@ -619,6 +621,8 @@ public function end($status = 0, $response = null)
             throw new ExitException($status);
         }
 
+        $this->flushLogger();
```
Comment (samdark) [inline src/base/Application.php]: I think it was meant to be flushed automatically even if app fails bad i.e. via register_shutdown_function.

Comment (samdark) [reply src/base/Application.php]: See here: https://github.com/yiisoft/yii2/blob/da447fd1bb9778b8b76994f3e3e14f9748c67625/framework/log/Logger.php#L122

**Comment by samdark** [comment]
Well, there are two ways handling it:

1. Register shutdown handler in the app and call logger there.
2. Register shutdown handler in the logger itself.

While `1.` was how it was done before, I think `2.` makes more sense since logging isn't the only thing done on shutdown.

**Comment by pgaultier** [comment]
Totally agree with you but as @hiqsol told me, `yiisoft/yii-core` classes cannot be used in `yiisoft/log` it will need a big rewrite of `yiisoft/log` with probably some duplicate code from `yiisoft/yii-core` and `yiisoft/di`

So right now, I think you can close this PR which is useless

**Comment by samdark** [comment]
> yiisoft/yii-core classes cannot be used in yiisoft/log

Correct but the other way around should be OK.

**Comment by hiqsol** [comment]
Please see https://github.com/yiisoft/log/pull/2

**Pull request closed without merging**